File: 404-sum-of-left-leaves/404-sum-of-left-leaves.cpp
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {
        int summ = 0;
        
        return travel(root, false );
        
    }
    
    int travel(TreeNode* head, bool is_left){
        if (head==NULL) return 0;
        
        if (is_left && head->left==NULL && head->right==NULL) return head->val;
        
        return travel(head->left,true)+travel(head->right,false);
        
        
    }
};

File: 404-sum-of-left-leaves/404-sum-of-left-leaves.py

class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        
        if not root:
            return 0
        
        def travel(node, flag):
            if not node:
                return 0

            if (flag and not node.left and not node.right):
                    return node.val

            return travel(node.left, True) + travel(node.right, False)
        
        return travel(root,False)
